core/src/main/scala/cromwell/util/JsonEditor.scala

```diff
@@ -0,0 +1,107 @@
+package cromwell.util
+
+import cats.data.NonEmptyList
+import io.circe.ACursor
+import io.circe.Json
+import mouse.all._
+import cats.syntax.traverse._
+import cats.syntax.functor._
+import cats.instances.list._
+import cats.data.State
+
+object JsonEditor {
+
+  def includeExcludeJson(json: Json, includeKeys: Option[NonEmptyList[String]], excludeKeys: Option[NonEmptyList[String]]): Json = {
+    (includeKeys, excludeKeys) match {
+      // Take includes, and then remove excludes
+      case (Some(includeKeys), Some(excludeKeys)) => includeJson(json, includeKeys) |> (excludeJson(_, excludeKeys))
```

Comment:
Does it make any sense to process the exclusions first, so that the inclusions are working with a smaller chunk of data? (I could be interpreting this completely wrong.)

Reply:
One or the other has to walk the whole tree, I thought includes would leave a smaller chunk than excludes would.